src/main/java/seedu/address/model/datetime/RecurringDateTime.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.model.datetime;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDateTime;
+
+/**
+ * Represents the frequency of a recurring appointment
+ */
+public class RecurringDateTime {
+
+    public static final String MESSAGE_CONSTRAINTS = "Recurring " + DateTime.MESSAGE_CONSTRAINTS_BODY;
+
+    private final long days;
+    private final long months;
+    private final long years;
+    private final long hours;
+    private final long minutes;
+
+    /**
+     * Constructs an {@code RecurringDateTime}.
+     *
+     * @param days, months, years, hours, minutes
+     */
+    public RecurringDateTime(long days, long months, long years, long hours, long minutes) {
```

Comment:
Will then all these be necessary fields for the user to input when creating a recurring task? Getting the user to input minutes seems to be a bit of a stretch.

Reply:
My rationale for this was say nurse wants to have a recurring appointment every 5 hours and 30 minutes or something? Then it would be convenient to have minutes in place. Anyway, we can keep this as it is for now and later if we find it irrelevant can remove it.